api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3589,23 +3590,62 @@ public void setObjectRetention(
       throws ErrorResponseException, IllegalArgumentException, InsufficientDataException,
           InternalException, InvalidBucketNameException, InvalidKeyException,
           InvalidResponseException, IOException, NoSuchAlgorithmException, XmlParserException {
-    if (config == null) {
+
+    this.setObjectRetention(
+        SetObjectRetentionArgs.builder()
+            .bucket(bucketName)
+            .object(objectName)
+            .versionId(versionId)
+            .config(config)
+            .bypassGovernanceRetention(bypassGovernanceRetention)
+            .build());
+  }
+
+  /**
+   * Sets retention configuration to an object.
+   *
+   * <pre>Example:{@code
+   * Retention retention =
+   *     new Retention(RetentionMode.COMPLIANCE, ZonedDateTime.now().plusYears(1));
+   * minioClient.setObjectRetention(SetObjectRetentionArgs args);
+   * }</pre>
+   *
+   * @param args {@link SetObjectRetentionArgs} object.
+   * @throws ErrorResponseException thrown to indicate S3 service returned an error response.
+   * @throws IllegalArgumentException throws to indicate invalid argument passed.
+   * @throws InsufficientDataException thrown to indicate not enough data available in InputStream.
+   * @throws InternalException thrown to indicate internal library error.
+   * @throws InvalidBucketNameException thrown to indicate invalid bucket name passed.
+   * @throws InvalidKeyException thrown to indicate missing of HMAC SHA-256 library.
+   * @throws InvalidResponseException thrown to indicate S3 service returned invalid or no error
+   *     response.
+   * @throws IOException thrown to indicate I/O error on S3 operation.
+   * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
+   * @throws XmlParserException thrown to indicate XML parsing error.
+   */
+  public void setObjectRetention(SetObjectRetentionArgs args)
+      throws ErrorResponseException, IllegalArgumentException, InsufficientDataException,
+          InternalException, InvalidBucketNameException, InvalidKeyException,
+          InvalidResponseException, IOException, NoSuchAlgorithmException, XmlParserException {
+    if (args.config() == null) {
```

Comment:
Its not required to check individual args (it defeats the purpose of arg builder). You would need to throw `IllegalArgumentException` `if (args == null)`